In [3]:
import pandas as pd
import numpy as np
#import lec2_graphs as gfx
import matplotlib.pyplot as plt
import scipy.stats as stats
plt.style.use('seaborn-whitegrid') # Gráficos estilo seaborn
plt.rcParams["figure.figsize"] = (4, 6) # Tamaño gráficos
plt.rcParams["figure.dpi"] = 100 # resolución gráficos

In [4]:
df = pd.read_csv('got_battles.csv')

In [5]:
df.head()

name  year  battle_number  \
0     Battle of the Golden Tooth   298              1   
1    Battle at the Mummer's Ford   298              2   
2             Battle of Riverrun   298              3   
3       Battle of the Green Fork   298              4   
4  Battle of the Whispering Wood   298              5   

              attacker_king             defender_king attacker_1 attacker_2  \
0  Joffrey/Tommen Baratheon                Robb Stark  Lannister        NaN   
1  Joffrey/Tommen Baratheon                Robb Stark  Lannister        NaN   
2  Joffrey/Tommen Baratheon                Robb Stark  Lannister        NaN   
3                Robb Stark  Joffrey/Tommen Baratheon      Stark        NaN   
4                Robb Stark  Joffrey/Tommen Baratheon      Stark      Tully   

  attacker_3 attacker_4 defender_1  ... major_death  major_capture  \
0        NaN        NaN      Tully  ...         1.0            0.0   
1        NaN        NaN  Baratheon  ...         1.0            0.0   
2        NaN        NaN      Tully  ...         0.0            1.0   
3        NaN        NaN  Lannister  ...         1.0            1.0   
4        NaN        NaN  Lannister  ...         1.0            1.0   

   attacker_size defender_size  \
0        15000.0        4000.0   
1            NaN         120.0   
2        15000.0       10000.0   
3        18000.0       20000.0   
4         1875.0        6000.0   

                                  attacker_commander  \
0                                    Jaime Lannister   
1                                     Gregor Clegane   
2                       Jaime Lannister, Andros Brax   
3  Roose Bolton, Wylis Manderly, Medger Cerwyn, H...   
4                          Robb Stark, Brynden Tully   

                                  defender_commander  summer         location  \
0                               Clement Piper, Vance     1.0     Golden Tooth   
1                                   Beric Dondarrion     1.0    Mummer's Ford   
2                      Edmure Tully, Tytos Blackwood     1.0         Riverrun   
3  Tywin Lannister, Gregor Clegane, Kevan Lannist...     1.0       Green Fork   
4                                    Jaime Lannister     1.0  Whispering Wood   

            region note  
0  The Westerlands  NaN  
1   The Riverlands  NaN  
2   The Riverlands  NaN  
3   The Riverlands  NaN  
4   The Riverlands  NaN  

[5 rows x 25 columns]

In [6]:
df.columns

Index(['name', 'year', 'battle_number', 'attacker_king', 'defender_king',
       'attacker_1', 'attacker_2', 'attacker_3', 'attacker_4', 'defender_1',
       'defender_2', 'defender_3', 'defender_4', 'attacker_outcome',
       'battle_type', 'major_death', 'major_capture', 'attacker_size',
       'defender_size', 'attacker_commander', 'defender_commander', 'summer',
       'location', 'region', 'note'],
      dtype='object')

In [7]:
df.shape

(38, 25)

In [8]:
df['attacker_king'].value_counts()

Joffrey/Tommen Baratheon    14
Robb Stark                  10
Balon/Euron Greyjoy          7
Stannis Baratheon            5
Name: attacker_king, dtype: int64

Observamos que los Baratheon concentran 14 de 38 batallas, seguidos por los Starks. Stannis
Baratheon, el "tío" de Joffrey, se posición como el que generó menos batallas con 5 de 35.

In [9]:
df['attacker_king'].value_counts() / len(df)

Joffrey/Tommen Baratheon    0.368421
Robb Stark                  0.263158
Balon/Euron Greyjoy         0.184211
Stannis Baratheon           0.131579
Name: attacker_king, dtype: float64

Se observa que entre Joffrey y Tomen Baratheon se concentra el 37% de las batallas, seguidas por
un 26% de Robb Stark.
la probabilidad que un Baratheon inicie una batalla es del 50%

Un ejemplo de ésto es la probabilidad que se produzca una muerte importante (medida con la
variable major_death )

In [10]:
df['major_death'].value_counts() / len(df)

0.0    0.631579
1.0    0.342105
Name: major_death, dtype: float64

La información solicitada es que hay un 34% de probabilidad que en una batalla cualquiera muera
un personaje importante.

## Contando unión e intersección de eventos